Resolve snyk warnings and expand model options (#204)
mckinsey/vizro: vizro-ai/docs/pages/user_guides/model_config.md

### OpenAI models

- gpt-3.5-turbo-0613 (to be deprecated on June 13, 2024)
- gpt-4-0613
- gpt-3.5-turbo-1106 (under testing)
- gpt-4-1106-preview (under testing, not suitable for production use)

These models provide different levels of performance and

> cost. In general, `gpt-3.5-turbo` collection provides the most cost-effective models,
> which would be a good starting point for most users. `gpt-4` is more powerful than the other, like
> it allows for more tokens per request. You can refer to these models' [capabilities](https://platform.openai.com/docs/models/overview)

Reviewer: ```suggestion
cost. In general, the `gpt-3.5-turbo` collection provides the most cost-effective models,
which would be a good starting point for most users. `gpt-4` models are more powerful than `gpt-3` models, e.g. they allow for more tokens per request. You can refer to these models' [capabilities](https://platform.openai.com/docs/models/overview)
```
